lib/scope.js

```diff
@@ -841,116 +850,182 @@ module.exports = {
     * @param field[n].tag {str} - HTML tag of node
     * @param field[n].type {str} - HTML type of node if it has one
     * @param field[n].guesses {arr} - arr of objects that could
-    *    possibly match a table row
+    *    possibly match a table row's `var` column
     * @param field[n].guesses[n].var {str} - name of variable
     * @param field[n].guesses[n].value {str} - value of a choice
     * @param <field[n].guesses[n].trigger> {str} - var name that triggered this page
     * 
     * @returns [ var_data[n] ]
     */
-    let matches = [];
-    let uses_proxies = false;
 
-    // There are mutliple guesses because field names are encoded
+    let rows_that_match_field_names = [];
+    let final_matching_rows = [];
```

Comment:
This var isn't used for about 100 lines, I'd move it closer to the relevant code.

Reply:
I'll move it as far down as I can, though it has to be kept in the outermost scope. In js, you have to define things outside of `if` statements or they later aren't defined in the scope outside of the `if` brackets.